rustler/src/types/binary.rs

```diff
@@ -1,43 +1,105 @@
+//! Safe wrappers around Erlang binaries.
+//!
+//! Rustler provides two types for wrapping Erlang binaries: [`Binary`] and
+//! [`OwnedBinary`].  Both represent a contiguous region `u8`s, and they both use
+//! the Erlang allocator. The primary difference between the two is ownership
+//! semantics.
+//!
+//! The _owned_ in `OwnedBinary` refers to the fact that it owns the VM binary is
+//! wraps, and the _owner_ of an `OwnedBinary` is free to modify its contents
+//! without fear of access violations or upsetting the VM. Ownership lasts until it
+//! is dropped by either falling out of scope, or converting it into a regular
+//! `Binary` via [`to_owned`] or one of the conversion traits. Once dropped, the VM
+//! is allowed to the free the backing memory at its leisure. Similar to `Vec`,
```

Comment:
This is not quite true.

1. I'd use "dropped" only for `Drop::drop` (as commented above)
2. While the NIF is still executing, an owned binary that was converted to a `Binary` does not free its backing store, it will still be a the exact same place but is considered read-only

I think I'd just remove that line entirely, the semantics of the types capture this nicely. Maybe replace it by something like

    A `Binary` belongs to a particular `Env`, so it can only be used for the duration of the remaining NIF call.